cnd/src/network.rs

```diff
@@ -444,45 +480,101 @@ impl libp2p::swarm::NetworkBehaviourEventProcess<::comit::network::comit::Behavi
                         tracing::error!("{}", e);
                     }));
             }
-            ::comit::network::comit::BehaviourOutEvent::OrderTaken {
-                order,
+        }
+    }
+}
+
+impl libp2p::swarm::NetworkBehaviourEventProcess<announce::BehaviourOutEvent> for ComitNode {
+    fn inject_event(&mut self, event: announce::BehaviourOutEvent) {
+        match event {
+            announce::BehaviourOutEvent::Confirmed {
                 peer,
-                refund_identity,
-                redeem_identity,
-                io,
+                shared_swap_id,
+                local_swap_id,
             } => {
-                tracing::info!("order taken: {:?}", order.id);
-                let local_swap_id = LocalSwapId::default();
-                let shared_swap_id = SharedSwapId::default();
+                let data = match self.local_data.remove(&local_swap_id) {
+                    Some(local_data) => local_data,
+                    None => {
+                        tracing::warn!("inconsistent data, missing local-data mapping");
+                        return;
+                    }
+                };
 
+                self.comit.communicate(peer, shared_swap_id, data);
                 self.local_swap_ids.insert(shared_swap_id, local_swap_id);
+            }
+            announce::BehaviourOutEvent::Failed {
+                peer,
+                local_swap_id,
+            } => {
+                tracing::warn!(
+                    "failed to complete announce protocol for swap {} with {}",
+                    local_swap_id,
+                    peer,
+                );
+            }
+        }
+    }
+}
 
-                let role = Role::Bob;
+impl libp2p::swarm::NetworkBehaviourEventProcess<orderbook::BehaviourOutEvent> for ComitNode {
+    fn inject_event(&mut self, event: orderbook::BehaviourOutEvent) {
+        match event {
+            orderbook::BehaviourOutEvent::TakeOrderRequest {
```

Comment:
I preferred the design where we wrapped the response channel and relevant data into a TakeOrderRequest struct that was passed into bob.confirm(). Bob is confirming a take order request. I think the channel here is not bobs concern and should be hidden.

Reply:
Putting the channel into a struct does not hide it from Bob. The struct is still an argument to the `confirm()` method so Bob has full knowledge of it. Also the current code is uniform with the `Announce` protocol. Uniformity is a good thing. By all means flag this somewhere (e.g. in a github issue) if you would like to see _both_ instances of this changed.

Reply:
we can made the channel private. I think that would hide it. Happy to tackle in an issue later to ensure uniformity